### ISSUE TITLE
Matching body as bytes

### ISSUE THREAD

**Issue by ErwanDL** [ggriffiniii/httptest#17]
Hello, is it possible to match an expected body composed of bytes, instead of a string or JSON like in the examples ?

I'd like to do something like that but it doesn't work : 

```rust
let bytes: Vec<u8> = vec![1, 2, 3, 4];
server.expect(
    Expectation::matching(request::body(bytes).respond_with(responders::status_code(200))
);
```

**Comment by ggriffiniii** [comment]
yes, there is a matcher defined for `&[u8]` to match an exact sequence of bytes. You just need to provide the bytes as a slice rather than a `Vec<u8>` to the `request::body` matcher. The `as_slice` method is probably the most straightforward way for your example.

```rust
let bytes: Vec<u8> = vec![1, 2, 3, 4];
server.expect(
    Expectation::matching(request::body(bytes.as_slice()).respond_with(responders::status_code(200))
);
```

**Comment by ErwanDL** [comment]
When I try to do that, I get an error from the borrow checker : "`bytes` does not live long enough: cast requires that `bytes` is borrowed for `'static`". Only way I've found to make it work is by declaring `bytes` as a `static` array, but that is very cumbersome for unit tests. Is it not possible to circumvent this lifetime requirement (I'm not sure what implementation constraints justify its existence) ?

**Comment by ggriffiniii** [comment]
You are correct that using a slice without a 'static lifetime will lead to borrow checker issues. That's a legitimate reason to pass ownership of the bytes into the matcher that I hadn't previously considered. I've added a matcher impl for `Vec<u8>` now and released a v0.15.4 that includes it. If you're able to upgrade to the latest version your original code snippet should work. Thanks for bringing this issue up and thanks for using httptest. I'm going to close the issue, but feel free to reopen if you still have a problem with the latest version.

**Comment by ErwanDL** [comment]
Thank you very much for the new matcher! And thank you for your work on this project and your reactivity although I believe you do this on your own free time, it's much appreciated.
